server/src/main/java/org/apache/druid/segment/loading/StorageLocationSelectorStrategy.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.loading;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.google.common.collect.ImmutableList;
+import org.apache.druid.timeline.DataSegment;
+
+import java.util.Iterator;
+
+/**
+ * This interface describes the storage location selection strategy which is responsible for ordering the
+ * available multiple {@link StorageLocation}s for optimal segment distribution.
```

Comment:
nit: I'm not sure the result segment distribution of any strategy can be called "optimal". If it's optimal, how is it optimal in terms of what metric? I would suggest to just remove "optimal" here.

Reply:
done